pkg/plugin/admissioncontrollers/webhook/mutating_controller.go

```diff
@@ -95,6 +94,48 @@ func (c *admissionController) mutate(ar *v1beta1.AdmissionReview) *v1beta1.Admis
 	}
 }
 
+func updateSchedulerName(patch []patchOperation) []patchOperation {
+	log.Logger.Info("updating scheduler name")
+	return append(patch, patchOperation{
+		Op:    "add",
+		Path:  "/spec/schedulerName",
+		Value: conf.GetSchedulerConf().SchedulerName,
+	})
+}
+
+func updateLabels(pod *v1.Pod, patch []patchOperation) []patchOperation {
+	log.Logger.Info("updating pod labels")
+	existingLabels := pod.Labels
+	result := make(map[string]string)
+	for k,v := range existingLabels {
+		result[k] = v
+	}
+
+	if _, ok := existingLabels[common.SparkLabelAppId]; !ok {
+		if _, ok := existingLabels[common.LabelApplicationId]; !ok {
+			// if app id not exist, generate one
+			generatedId := fmt.Sprintf("autogen_%s_%s", pod.Name, uuid.New().String())
+			log.Logger.Debug("adding application ID",
+				zap.String("generatedID", generatedId))
+			result[common.LabelApplicationId] = generatedId
+		}
+	}
```

Comment:
setting an app ID I would like to see configurable or turned off by default in the code we generate with a default build.

Reply:
The default build will not include admission controller, including our helm charts. The admission controller should be limited to testing, when we want to take over a cluster without asking apps changing their code. However, once we reach our testing goals, we should turn the admission controller off.

When this is enabled. We have to create app ID when **it is not set**. Otherwise, pods will be pending.

Reply:
OK, if this is just for test cases I am OK